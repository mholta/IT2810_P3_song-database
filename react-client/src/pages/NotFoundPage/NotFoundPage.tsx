import React from 'react';
import { useHistory, useLocation } from 'react-router';
import { SectionsWrapper } from '../../components/elements/Section';
import { Routes } from '../MainRouter';

/**
 * Page for when we enter a route that does not lead to any other page.
 */
const NotFoundPage = () => {
  const location = useLocation();
  const history = useHistory();

  if (location.pathname !== Routes.NOT_FOUND) history.replace(Routes.NOT_FOUND);

  return (
    <SectionsWrapper>Side ikke funnet. Prøv å søk på noe.</SectionsWrapper>
  );
};

export default NotFoundPage;
