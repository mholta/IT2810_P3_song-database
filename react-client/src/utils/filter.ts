/**
 * Compares two booleans for sorting.
 */
export const compareBoolean = (a: boolean, b: boolean) => {
  if (a && !b) return -1;
  if (!a && b) return 1;
  return 0;
};
