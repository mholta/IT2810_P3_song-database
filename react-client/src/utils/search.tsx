export const getQueryStringFromFormSubmitEvent = (
  formData: FormData
): string => {
  return new URLSearchParams(formData as URLSearchParams).toString();
};
