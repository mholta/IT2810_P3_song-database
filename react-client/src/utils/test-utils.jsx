// Copied and modified from https://redux.js.org/usage/writing-tests

import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from '../store/index';
import PageLayoutWrapper from '../components/layout/PageLayoutWrapper';
import { ThemeProvider } from '@mui/material';
import { theme } from '../styles/theme';
import { MemoryRouter, Route } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';

function render(
  ui,
  gqlMocks = {},
  url = '/',
  params = {},
  {
    preloadedState,
    store = createStore(rootReducer, composeWithDevTools()),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    let initEntry = url;
    for (let key of Object.keys(params)) {
      initEntry = initEntry.replace(':' + key, params[key]);
    }

    return (
      <MockedProvider mocks={gqlMocks}>
        <Provider store={store}>
          <ThemeProvider theme={theme}>
            <MemoryRouter initialEntries={[initEntry]}>
              <PageLayoutWrapper>
                <Route path={url}>{children}</Route>
              </PageLayoutWrapper>
            </MemoryRouter>
          </ThemeProvider>
        </Provider>
      </MockedProvider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };
