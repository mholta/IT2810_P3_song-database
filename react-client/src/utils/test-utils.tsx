import React from 'react';
import { Provider } from 'react-redux';
import { CombinedState, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from '../store/index';
import PageLayoutWrapper from '../components/layout/PageLayoutWrapper';
import { ThemeProvider } from '@mui/material';
import { theme } from '../styles/theme';
import { MemoryRouter, Route } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { LayoutState } from '../store/layout/layout.reducer';
import { FilterState } from '../store/filter/filter.reducer';

interface TestProps {
  children: any;
  gqlMocks?: any;
  url: string;
  params?: any;
  isPage: boolean;
  store?: any;
}

export const createTestStore = () => {
  return createStore(rootReducer, composeWithDevTools());
};

export const TestProvider = ({
  children,
  gqlMocks = [],
  url,
  params = {},
  isPage = true,
  store,
}: TestProps) => {
  let initEntry = url;
  for (let key of Object.keys(params)) {
    initEntry = initEntry.replace(':' + key, params[key]);
  }

  if (!store) {
    store = createTestStore();
  }

  return (
    <MockedProvider mocks={gqlMocks}>
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <MemoryRouter initialEntries={[initEntry]}>
            {isPage && (
              <PageLayoutWrapper>
                <Route path={url}>{children}</Route>
              </PageLayoutWrapper>
            )}
            {!isPage && <Route path={url}>{children}</Route>}
          </MemoryRouter>
        </ThemeProvider>
      </Provider>
    </MockedProvider>
  );
};
