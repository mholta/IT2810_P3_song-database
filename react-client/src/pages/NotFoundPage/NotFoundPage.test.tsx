import { TestProvider } from '../../utils/test-utils';
import renderer from 'react-test-renderer';
import NotFoundPage from './NotFoundPage';

describe('NotFoundPage', () => {
  test('rendered successfully', () => {
    const tree = renderer
      .create(
        <TestProvider url="/404">
          <NotFoundPage />
        </TestProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
