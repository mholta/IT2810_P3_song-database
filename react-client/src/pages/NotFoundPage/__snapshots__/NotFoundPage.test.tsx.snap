// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage rendered successfully 1`] = `
<div
  className="css-q8cb76"
>
  <div
    className="css-144yc9n"
  >
    <nav
      className="css-ag1rcq"
      open={false}
    >
      <div
        className="css-ydsbw5"
      >
        <a
          className="css-136ox9k"
          href="/search"
          onClick={[Function]}
        >
           
          <span>
            Sanger
          </span>
        </a>
        <a
          className="css-136ox9k"
          href="/submit-song"
          onClick={[Function]}
        >
           
          <span>
            Send inn sang
          </span>
        </a>
      </div>
    </nav>
  </div>
  <div
    style={
      Object {
        "flexGrow": 1,
      }
    }
  >
    <div
      className="css-kbek2b"
    >
      <div
        className="css-0"
      >
        <form
          action="/search"
          className="css-ofpocj"
          onSubmit={[Function]}
        >
          <div
            className="css-1rmzhqg"
          >
            <div
              className="css-1f7apd6"
            >
              <div
                className="css-18a5p6x"
                focused={0}
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </div>
              <input
                autoComplete="off"
                className="css-psh2zg"
                name="query"
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                open={0}
                placeholder="Søk på sang"
                type="search"
                value=""
              />
            </div>
            <div
              aria-label="hamburger menu button"
              className="css-macrnf"
              data-testid="hamburger-button"
              onClick={[Function]}
              open={false}
            >
              <span />
              <span />
              <span />
              <span />
            </div>
          </div>
        </form>
      </div>
    </div>
    <main>
      <div
        className="css-2x3ewd"
      >
        <div
          className="css-2x3ewd"
        >
          Side ikke funnet. Prøv å søk på noe.
        </div>
      </div>
    </main>
  </div>
</div>
`;
